Jint/Runtime/Interpreter/Expressions/JintExpression.cs

```diff
@@ -52,6 +55,40 @@ protected virtual void Initialize()
 
         protected abstract object EvaluateInternal();
 
+        /// <summary>
+        /// If we'd get Esprima source, we would just refer to it, but this makes error messages easier to decipher.
+        /// </summary>
+        internal string SourceText => ToString(_expression) ?? "*unknown*";
+
+        internal static string? ToString(Expression expression)
+        {
+            while (true)
+            {
+                if (expression is Literal literal)
+                {
+                    return EsprimaExtensions.LiteralKeyToString(literal);
+                }
+
+                if (expression is Identifier identifier)
+                {
+                    return identifier.Name;
+                }
+
+                if (expression is MemberExpression memberExpression)
+                {
+                    return ToString(memberExpression.Object) + "." + ToString(memberExpression.Property);
+                }
+
+                if (expression is CallExpression callExpression)
+                {
+                    expression = callExpression.Callee;
+                    continue;
```

Comment:
The `while(true)` is only for this case? not a recursion instead?

Reply:
This is honestly a hack. The correct way to handle this later is to have parser enable `options.Range = true` which should give us start and end index, then if we have the script (string instance), we can substring it. Might even make sense to always provide those out of Esprima without config switch if there's  no real performance penalty. So this is something like best-effort, should be replaced with proper solution...